src/applications/vre/definitions/profileAddress.js

```diff
@@ -0,0 +1,265 @@
+      },
+      city: {
+        'ui:required': callback,
+        'ui:errorMessages': {
+          required: 'City is required',
+          pattern: 'City must be under 100 characters',
+        },
+        'ui:options': {
+          replaceSchema: (formData, schema, uiSchema, index) => {
+            const formDataPath = getPath(path, index);
+            const { isMilitary } = get(formDataPath, formData);
+            if (isMilitary) {
+              return {
+                type: 'string',
+                title: 'APO/FPO/DPO',
+                enum: militaryCities.map(city => city.value),
+                enumNames: militaryCities.map(city => city.label),
+              };
+            }
+            return {
+              type: 'string',
+              title: 'City',
+              minLength: 1,
+              maxLength: 100,
+              pattern: '^.*\\S.*',
+            };
+          },
+        },
+      },
+      state: {
+        'ui:required': (formData, index) => {
+          // Only required if the country is the United States;
+          const formDataPath = getPath(path, index);
+          const { country } = get(formDataPath, formData);
+          return country && country === USA.value;
+        },
+        'ui:errorMessages': {
+          required: 'Please enter a valid State, Province, or Region',
+        },
+        'ui:options': {
+          replaceSchema: (formData, schema, uiSchema, index) => {
+            const formDataPath = getPath(path, index);
+            const { country, isMilitary } = get(formDataPath, formData);
+            if (isMilitary) {
+              return {
+                type: 'string',
+                title: 'State',
+                enum: Object.keys(MILITARY_STATES),
+                enumNames: Object.values(MILITARY_STATES),
+              };
+            } else if (!isMilitary && country === 'USA') {
+              return {
+                type: 'string',
+                title: 'State',
+                enum: states50AndDC.map(state => state.value),
+                enumNames: states50AndDC.map(state => state.label),
+              };
+            } else {
+              return {
+                type: 'string',
+                title: 'State/Province/Region',
+              };
+            }
+          },
+        },
+      },
+      postalCode: {
+        'ui:required': callback,
+        'ui:title': 'Postal Code',
+        'ui:errorMessages': {
+          required: 'Postal code is required',
+          pattern: 'Please enter a valid US zip code',
+        },
+        'ui:options': {
+          widgetClassNames: 'usa-input-medium',
+          replaceSchema: (formData, schema, uiSchema, index) => {
+            const formDataPath = getPath(path, index);
+            const { country, isMilitary } = get(formDataPath, formData);
+            if (isMilitary || country === 'USA') {
+              return {
+                type: 'string',
+                pattern: '(\\d{5})(?:[-](\\d{4}))?$',
```

Comment:
I've always found dropping a comment on regex patterns helpful for understanding the intent and reason later.

Reply:
I totally understand the reason for that regex pattern 

<img alt="ghe-emoji" title=":_tuzki_dizzy:" src="https://cloud.githubusercontent.com/assets/136959/13203535/61480e08-d880-11e5-9ea8-7fc03c7d7b56.gif">